### PR TITLE
CR#26693 Allow multiple partitions

### PR DESCRIPTION
**Pull request by DreierF**
Addresses issue [TS-26693](https://cqse.atlassian.net/browse/TS-26693)

- [ ] Changes are tested adequately
- [x] Agent's README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)
- [x] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [x] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: system-tests/default-excludes-test/src/test/java/com/teamscale/tia/client/TeamscaleMockServer.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 	public TeamscaleMockServer(int port) throws IOException {
 		port(port);
-		post("api/projects/:projectName/external-analysis/session/auto-create/report", this::handleReport);
+		post("api/v5.9.0/projects/:projectName/external-analysis/session/auto-create/report", this::handleReport);
```
Comment (VitaNuova) [inline system-tests/default-excludes-test/src/test/java/com/teamscale/tia/client/TeamscaleMockServer.java:33]: Why are all these methods calling the APIs of specific Teamscale versions?

File: teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java [hunk 2]
```diff
@@ -81,33 +81,32 @@
...
 	/** Retrieve clustered impacted tests based on the given available tests and baseline timestamp. */
-	@PUT("api/projects/{projectName}/impacted-tests")
+	@PUT("api/v6.5.2/projects/{projectName}/impacted-tests?prioritization-strategy=CHEAP_ADDITIONAL_COVERAGE_PER_TIME")
 	Call<List<PrioritizableTestCluster>> getImpactedTests(
 			@Path("projectName") String projectName,
-			@Query("baseline") Long baseline,
+			@Query("baseline") String baseline,
 			@Query("end") CommitDescriptor end,
-			@Query("partitions") String partition,
+			@Query("partitions") List<String> partitions,
```
Comment (VitaNuova) [inline teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java:89]: Did you change the corresponding Teamscale service query option, or did it already support multiple partitions?

File: teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java [hunk 3]
```diff
@@ -81,33 +81,32 @@
 	);
 
 	/** Retrieve clustered impacted tests based on the given available tests and baseline timestamp. */
-	@PUT("api/projects/{projectName}/impacted-tests")
+	@PUT("api/v6.5.2/projects/{projectName}/impacted-tests?prioritization-strategy=CHEAP_ADDITIONAL_COVERAGE_PER_TIME")
```
Comment (VitaNuova) [inline teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java:84]: Is it fine that you're adding this prioritization strategy option to all calls to this endpoint, although it's only necessary for the current customer? Might we need another endpoint without this option, or might we need to parametrize this endpoint to be able to call with and without this strategy option?

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,9 @@ We use [semantic versioning](http://semver.org/):
 - PATCH version when you make backwards compatible bug fixes.
 
 # Next Release
+- [breaking change] _teamscale-client_: Impacted tests now allows to specify multiple partitions
+- [breaking change] _teamscale-client_: Impacted tests baseline can now be a string
```
Comment (VitaNuova) [inline CHANGELOG.md:9]: What does this have to do with the customer request? And what's the benefit? Is that that you're now able to pass baseline names, like "Baseline Sprint 25"?

File: system-tests/default-excludes-test/src/test/java/com/teamscale/tia/client/TeamscaleMockServer.java (hunk 1 above)
Comment (DreierF) [reply system-tests/default-excludes-test/src/test/java/com/teamscale/tia/client/TeamscaleMockServer.java:33]: Because otherwise we will break the agent once we introduce a breaking change to the API in the future. Instead when all external tools specify the version they expect (like the Java based IDEs already do) we can safely change stuff in the future without having to way years until all tools and instances are updated (as we previously had to)

Comment (DreierF) [reply system-tests/default-excludes-test/src/test/java/com/teamscale/tia/client/TeamscaleMockServer.java:33]: I mean isn't that the whole point of versioned APIs

File: teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java (hunk 2 above)
Comment (DreierF) [reply teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java:89]: It already did. We just didn't expose it here yet

File: teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java (hunk 3 above)
Comment (DreierF) [reply teamscale-client/src/main/java/com/teamscale/client/ITeamscaleService.java:84]: This will become the new default soon. All other options are only for research purposes, but will not be used for production systems. So I thought it would be better to keep this as an implementation detail

File: CHANGELOG.md (hunk 4 above)
Comment (DreierF) [reply CHANGELOG.md:9]: I missed to add that to the ticket description, but that was also requested. It's mostly to allow specifying 123456p1 baselines, but also to allow HEAD for example
